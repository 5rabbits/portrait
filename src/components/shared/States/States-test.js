import { React, toJSON, mount } from 'test/helper'
import States from 'shared/States'

describe('States', () => {
  it('should render with minimum props', () => {
    const component = mount(<States />)

    expect(toJSON(component)).toMatchSnapshot()
  })
})
