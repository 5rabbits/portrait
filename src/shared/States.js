import React from 'react'
import { Loader, EmptyView } from 'shared'

const States = props =>
  <div className="section-states">
    <article className="hidden section-state" id="block-content-loading">
      <Loader />
    </article>
    <article className="hidden section-state" id="block-content-empty">
      <EmptyView>
        {props.empty}
      </EmptyView>
    </article>
    <article className="hidden- section-state" id="block-content-data">
      {props.content}
    </article>
  </div>

export default States
