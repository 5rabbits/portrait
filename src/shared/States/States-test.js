import { React, expect, mount } from 'test/helper'
import States from 'shared/States'

describe('States', () => {
  it('should render with minimum props', () => {
    const component = mount(<States />)

    expect(component).to.be.ok
  })
})
