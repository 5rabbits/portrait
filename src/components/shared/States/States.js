import React from 'react'
import { Loader, EmptyView } from 'shared'
import childrenPropType from 'propTypes/children'

const States = props =>
  <div className="section-states">
    <article className="hidden section-state" id="block-content-loading">
      <Loader />
    </article>
    <article className="hidden section-state" id="block-content-empty">
      <EmptyView>
        {props.empty}
      </EmptyView>
    </article>
    <article className="hidden- section-state" id="block-content-data">
      {props.content}
    </article>
  </div>

States.propTypes = {
  content: childrenPropType,
  empty: childrenPropType,
}

States.defaultProps = {
  content: null,
  empty: null,
}

export default States
