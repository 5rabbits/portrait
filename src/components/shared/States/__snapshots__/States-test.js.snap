// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`States should render with minimum props 1`] = `
<States
  content={null}
  empty={null}
>
  <div
    className="section-states"
  >
    <article
      className="hidden section-state"
      id="block-content-loading"
    >
      <Loader>
        <EmptyView>
          <div
            className="empty-view"
          >
            <div
              className="empty-container"
            >
              <div
                className="empty-dialog"
              >
                <i
                  className="fa fa-circle-o-notch fa-5x color-soft fa-spin"
                />
              </div>
            </div>
          </div>
        </EmptyView>
      </Loader>
    </article>
    <article
      className="hidden section-state"
      id="block-content-empty"
    >
      <EmptyView>
        <div
          className="empty-view"
        >
          <div
            className="empty-container"
          >
            <div
              className="empty-dialog"
            />
          </div>
        </div>
      </EmptyView>
    </article>
    <article
      className="hidden- section-state"
      id="block-content-data"
    />
  </div>
</States>
`;
